src/include/btmem.h

```diff
@@ -960,15 +946,71 @@ struct __wt_ref {
+     *
+     * Because ft_info is a union it is important to always access the correct field. It is also
+     * vital to interpret the state correctly and consider all the possible cases.
+     *
+     * The union access should be ft_info.del if the state is WT_REF_DELETED (states 1 and 2 below),
+     * and should be ft_info.update if the state is WT_REF_MEM (states 5-6 below). Otherwise,
+     * neither field is valid and the pointer should always be NULL.
+     *
+     * These are the possible states:
+     *
+     * 1. The WT_REF state is WT_REF_DELETED and ft_info.del is NULL. This means the page is deleted
+     * and the deletion is globally visible. Any on-disk page has been or will be discarded.
+     *
+     * 2. The WT_REF state is WT_REF_DELETED and ft_info.del is not NULL. The page is on disk, and
+     * is deleted, but the deletion may not yet be globally visible (or visible to any given reader
+     * either). ft_info.del describes the delete operation. If it is necessary to read the page on
+     * behalf of a thread that cannot see the deletion, the page must be instantiated as described
+     * above.
+     *
+     * 3. The WT_REF state is WT_REF_DISK, and the parent page's address cell is a deleted-address
+     * cell. ft_info is not valid; ft_info.del should read as NULL. The page is on disk, and
+     * deleted; the deletion may not yet be globally visible. Because the time aggregate stored in
+     * the parent internal page includes the deletion time, tree walks will skip the page as
+     * appropriate without needing the fast-delete information. This state can only happen in
+     * readonly trees; it is a result of the page being read in and instantiated, but not marked
+     * dirty, then discarded by eviction. (In principle eviction should set the state back to
+     * WT_REF_DELETED in this case; however, this turns out to be awkward.) The only time this state
+     * arises is when reading in the page, at which point we can check the address cell and retrieve
+     * the fast-delete information. Otherwise, it is indistinguishable from state 4.
+     *
+     * 4. The WT_REF state is WT_REF_DISK, and the parent page's address cell is not a
+     * deleted-address cell. ft_info is not valid; ft_info.del should read as NULL. This is an
+     * ordinary on-disk page.
+     *
+     * 5. The WT_REF state is WT_REF_MEM, and ft_info.update is NULL. This is an ordinary in-memory
+     * page.
+     *
+     * 6. The WT_REF state is WT_REF_MEM, and ft_info.update is not NULL.
+     * This is a deleted page that was instantiated when the delete transaction was not yet
+     * committed. ft_info.update is the list of updates created by the instantiation, which is used
+     * to commit or abort them as needed and then cleared. It is not possible to get to this state
+     * if the truncate information was read from disk; uncommitted (including prepared) truncates
+     * are not evicted or checkpointed.
```

Comment:
Comment needs formatting -- format script skips leading numbers, so have to do by hand.

Reply:
Whoops. Fixed in [42d20ae](https://github.com/wiredtiger/wiredtiger/pull/7909/commits/42d20ae234f880c06e2cc2133a5c772b2c7414a6).